api/lib/domain/services/schooling-registrations-xml-service.js

```diff
@@ -1,91 +1,147 @@
-    .map((xmlSchoolingRegistration) => {
-      const schoolingRegistrationStructuresContainer = _findSchoolingRegistrationStructures(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures);
-
-      const birthCountryCode = _getEitherElementValueOrNull(xmlSchoolingRegistration, 'CODE_PAYS');
-      return {
-        lastName: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'NOM_DE_FAMILLE'),
-        preferredLastName: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'NOM_USAGE'),
-        firstName: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'PRENOM'),
-        middleName: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'PRENOM2'),
-        thirdName: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'PRENOM3'),
-        birthdate: moment(_getEitherElementValueOrNull(xmlSchoolingRegistration, 'DATE_NAISS'), 'DD/MM/YYYY').format('YYYY-MM-DD'),
-        birthCountryCode,
-        birthProvinceCode: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'CODE_DEPARTEMENT_NAISS'),
-        birthCityCode: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'CODE_COMMUNE_INSEE_NAISS'),
-        birthCity: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'VILLE_NAISS'),
-        MEFCode: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'CODE_MEF'),
-        status: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'CODE_STATUT'),
-        nationalStudentId: _getEitherElementValueOrNull(xmlSchoolingRegistration, 'ID_NATIONAL'),
-        division: _findSchoolingRegistrationDivision(schoolingRegistrationStructuresContainer),
-      };
+async function extractSchoolingRegistrationsInformationFromSIECLE(payload, organization) {
+  const firstline = await _readFirstLineFromFile(payload.path);
+  const encoding = xmlEncoding(Buffer.from(firstline)) || DEFAULT_FILE_ENCODING;
+
+  return new Promise(function(resolve, reject) {
+    const fileStream = fs.createReadStream(payload.path).pipe(iconv.decodeStream(encoding));
+
+    const streamerToParseOrganizationUAI = new saxPath.SaXPath(saxParser, NODE_ORGANIZATION_UAI);
+    const streamerToParseSchoolingRegistrations = new saxPath.SaXPath(saxParser, NODES_SCHOOLING_RESGISTRATIONS);
+
+    const mapSchoolingRegistrationsByStudentId = new Map();
+    const nationalStudentIds = [];
+
+    streamerToParseOrganizationUAI.on('match', (xmlFormat) => {
+      parseString(xmlFormat, function(err, jsFormat) {
+        if (jsFormat.PARAMETRES) {
+          const UAIFromSIECLE = _getValueFromParsedElement(jsFormat.PARAMETRES.UAJ);
+          if (UAIFromSIECLE !== organization.externalId) {
+            reject(new FileValidationError(UAI_SIECLE_FILE_NOT_MATCH_ORGANIZATION_UAI));
+          }
+        }
+      });
     });
 
-  return { UAIFromSIECLE, resultFromExtraction };
-}
+    streamerToParseSchoolingRegistrations.on('match', (xmlFormat) => {
+      if (xmlFormat.startsWith(ELEVE_ELEMENT) || xmlFormat.startsWith(STRUCTURE_ELEVE_ELEMENT)) {
```

Comment:
C'est une question de préférence, mais j'aime bien essayer de rendre le code un poil plus expressif.
Déjà, je renommerais `xmlFormat` en `xmlNode`, puis je ferais :
```suggestion
    streamerToParseSchoolingRegistrations.on('match', (xmlNode) => {
      if (_isSchoolingRegistrationNode(xmlNode)) {
      /*...*/
      }
   }
  
  function _isSchoolingRegistrationNode(xmlNode) {
    return xmlNode.startsWith(ELEVE_ELEMENT) || xmlNode.startsWith(STRUCTURE_ELEVE_ELEMENT)
  }
```

Reply:
Done, j'aime bien aussi